resources/gradle_plugin/src/main/groovy/ConnextDdsCompileExample.groovy

```diff
@@ -0,0 +1,265 @@
+        project.getDependencies().add("implementation", ":nddsjava")
+
+        /** Adds and configures the created tasks */
+        this.addCodegenTask(project, extension)
+        this.addDeleteGeneratedFilesTask(project, extension)
+        this.addAndConfigureApplicationPlugin(project, extension)
+    }
+
+    /**@
+     * Sets the source directory with the path which contains the .java files.
+     *
+     * @param project the gradle's project object
+     * @param sourceDir the source directory to set
+     */
+    private void setSourceDir(Project project, String sourceDir)
+    {
+        def sourceSets = project
+            .getExtensions()
+            .getByType(SourceSetContainer.class)
+        def main = sourceSets.getByName(SourceSet.MAIN_SOURCE_SET_NAME)
+        main.getJava().srcDir(sourceDir)
+    }
+
+    /**@
+     * Adds the codegen task to the project. This task calls rtiddsgen to
+     * generate all the required source files for building the examples.
+     *
+     * @param project the gradle's project object
+     * @param extension contains the name of the idlFile and the codegen extra
+     * args
+     */
+    private void addCodegenTask(
+            Project project,
+            ConnextDdsBuildExampleExtension extension)
+    {
+        /** Adds the codegen task */
+        def codegen = project.task(CODEGEN_TASK_NAME) {
+            doFirst {
+                /** There are a few examples without .idl file */
+                if(extension.idlFile.get()) {
+                    def codegenExecutableName = "rtiddsgen"
+
+                    /** 
+                     * -ppDisable is required for compiling in Windows without
+                     * needing Visual Studio or Ninja. In the future it would
+                     * be convenient to delete this argument and create a new
+                     * task to find Visual Studio or Ninja.
+                     */
+                    def codegenArgs = [
+                        "-language",
+                        "Java",
+                        "-ppDisable",
+                        "-update",
+                        "typefiles",
+                        extension.idlFile.get()
+                    ]
+
+                    if (OperatingSystem.current().isWindows()) {
+                        codegenExecutableName += ".bat"
+                    }
+
+                    /** inserts the extra args if it was given */
+                    if (extension.codegenExtraArgs.get()) {
+                        codegenArgs.addAll(
+                            codegenArgs.size() - 1, extension.codegenExtraArgs.get())
+                    }
+
+                    /** execs rtiddsgen */
+                    project.exec {
+                        executable codegenExecutableName
+                        args codegenArgs
+                    }
+                } else {
+                    println "Skipping codegen because the .idl file was not provided"
```

Comment:
shouldn't this be an error/exception? no IDL file, no party

Reply:
The dynamic data examples don't use an IDL file.